jobs/build-charms/builder_local.py

```diff
@@ -577,17 +605,51 @@ def upload_args(self) -> Mapping[str, str]:
 
 @dataclass
 class Artifact:
-    arch: Arch
     charm_or_bundle: Path
+    arch: Arch = Arch.ALL
+    series: Series = Series.ALL
     rev: int = None  # set when uploaded
     resources: List[CharmResource] = field(default_factory=list)
 
+    def __str__(self) -> str:
+        return f"File: {self.charm_or_bundle.name} for arch={self.arch} and series={self.series}"
+
+    @staticmethod
+    def _from_run_on_base(run_on_base: str) -> Iterable[Tuple[Arch, Series]]:
+        if not run_on_base.startswith("ubuntu-"):
+            return
+        base, *archs = run_on_base.split("-")[1:]
+        for arch in archs:
+            yield Arch.from_value(arch), Series.from_value(base)
+
+    @classmethod
+    def from_charm(cls, charm_file: Path) -> ["Artifact"]:
+        """
+        Parsed according to charmcraft file output.
+        https://discourse.charmhub.io/t/charmcraft-bases-provider-support/4713
+        """
+        run_on_bases = charm_file.stem.split("_")[1:]
+        base_arches = list(
+            pair for each in run_on_bases for pair in Artifact._from_run_on_base(each)
+        )
+        arch, series = base_arches[0]
+        if len(base_arches) == 1:
+            # single series, single arch     --> Series._specific_, Arch._specific_
+            return cls(charm_file, arch, series)
+        if len(set(arch for arch, _ in base_arches)) == 1:
+            # multiple series, single arch   --> Series.ALL       , Arch._specific_
+            return cls(charm_file, arch, Series.ALL)
+        if len(set(series for _, series in base_arches)) == 1:
+            # single series, multiple arch   --> Series._specific_, Arch.ALL
+            return cls(charm_file, Arch.ALL, series)
+        return cls(charm_file, Arch.ALL, Series.ALL)
+
```

Comment:
uses the name of the charm to determine which Series/Arch it belongs to.  See test cases for details